### PR TITLE
Part15 all tasks done

### PR DESCRIPTION
**Pull request by alexkulesh**

### CODE REVIEW

File: FirstSolution/Part15Task1P2and3/Tram.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
+﻿using System;
+
+namespace Part15Task1P2
+{
+    public class Tram : Transport
+    {   
+        static Tram() => TransportType = "Tram";
```
Comment (Alex19777) [inline FirstSolution/Part15Task1P2and3/Tram.cs]: Use const

File: FirstSolution/Part15Task1P2and3/Transport.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+{
+    public class Transport
+    {
+        private static string transportType;
+        private string destination;
+        private int number;
+        private string dateOfDepature;
+        private int numberOfSeats;
+        
+        public Transport()
```
Comment (Alex19777) [inline FirstSolution/Part15Task1P2and3/Transport.cs]: No need to define empty default constructor

File: FirstSolution/Part15Task1P2and3/Transport.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
+﻿using System;
+
+namespace Part15Task1P2
+{
+    public class Transport
+    {
+        private static string transportType;
+        private string destination;
+        private int number;
+        private string dateOfDepature;
```
Comment (Alex19777) [inline FirstSolution/Part15Task1P2and3/Transport.cs]: Try to use DateTime type

File: FirstSolution/Part15Task1P2and3/Task.cs [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+                
+            for (int i = 0; i < transports.Length; i++)
+            {
+                if (input == transports[i].DateOfDepature)
+                {
+                    Console.WriteLine(transports[i]);
+                }
+            }
+            
+            for (int i = 0; i < transports.Length; i++)
```
Comment (Alex19777) [inline FirstSolution/Part15Task1P2and3/Task.cs:43]: Do you need to print TransportType here?

File: FirstSolution/Part15Task2/Car.cs [hunk 5]
```diff
@@ -0,0 +1,10 @@
+﻿namespace Part15Task2
+{
+    public class Car : Auto
+    {
```
Comment (Alex19777) [inline FirstSolution/Part15Task2/Car.cs:6]: Add unique class member

File: FirstSolution/Part15Task2/Motorcycle.cs [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+            
+            return base.countCarryingCapacity();
+        }
+
+        public Motorcycle(string brand, string registrationNum, int maxSpeed, int carryingCapacity, bool hasSidebar) 
+            : base(brand, registrationNum, maxSpeed, carryingCapacity)
+        {   
+            this.hasSidebar = hasSidebar;
+            countCarryingCapacity();
+        }
```
Comment (Alex19777) [inline FirstSolution/Part15Task2/Motorcycle.cs]: Constructor should be in front of methods, please change the members order. https://stackoverflow.com/questions/150479/order-of-items-in-classes-fields-properties-constructors-methods

File: FirstSolution/Part15Task3/Product.cs [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+    {
+        private string name;
+        private double price;
+        private DateTime dateOfProduction;
+        private DateTime bestBeforeDate;
+        private bool isExpired;
+        
+        public Product()
+        {
+        }
```
Comment (Alex19777) [inline FirstSolution/Part15Task3/Product.cs]: Delete empty constructor

File: FirstSolution/Part15Task3/Set.cs [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+            }
+        }
+        
+        public override void compareDates()
+        {
+            for (int i = 0; i < items.Length; i++)
+            {
+                if (items[i].BestBeforeDate > DateTime.Today)
+                {
+                    Console.WriteLine("Item is not expired");
```
Comment (Alex19777) [inline FirstSolution/Part15Task3/Set.cs]: Put some info about the item

File: FirstSolution/Part15Task2/Car.cs (hunk 5 above)
Comment (alexkulesh) [reply FirstSolution/Part15Task2/Car.cs:6]: Done

File: FirstSolution/Part15Task1P2and3/Transport.cs (hunk 2 above)
Comment (alexkulesh) [reply FirstSolution/Part15Task1P2and3/Transport.cs]: Changed

File: FirstSolution/Part15Task2/Motorcycle.cs (hunk 6 above)
Comment (alexkulesh) [reply FirstSolution/Part15Task2/Motorcycle.cs]: Sorry I forgot that, and quickly added that block of code. Just changed that.

File: FirstSolution/Part15Task3/Product.cs (hunk 7 above)
Comment (alexkulesh) [reply FirstSolution/Part15Task3/Product.cs]: Done

File: FirstSolution/Part15Task3/Set.cs (hunk 8 above)
Comment (alexkulesh) [reply FirstSolution/Part15Task3/Set.cs]: Done

File: FirstSolution/Part15Task1P2and3/Tram.cs (hunk 1 above)
Comment (alexkulesh) [reply FirstSolution/Part15Task1P2and3/Tram.cs]: Done

File: FirstSolution/Part15Task1P2and3/Transport.cs (hunk 3 above)
Comment (alexkulesh) [reply FirstSolution/Part15Task1P2and3/Transport.cs]: Done

**Review by Alex19777 — APPROVED** [review]

**Pull request merged**